src/unix/index_unix.ml

```diff
@@ -260,6 +260,34 @@ module IO : Index.IO = struct
       let _ = Unix.fstat t.raw.fd in
       true
     with Unix.Unix_error (Unix.EBADF, _, _) -> false
+
+  type lock = Unix.file_descr
+
+  let _lock op f =
```

Comment:
To be consistent with the rest of the module (and avoid ignored warning `32`), could you rename this to `unsafe_lock` for instance ?